relation_join.go

```diff
@@ -312,3 +344,64 @@ func appendChildValues(
 	}
 	return b
 }
+
+func getColumns(table schema.Safe, fields []*schema.Field) [][]byte {
+	//Based upon query_base.appendColumns
+	var list [][]byte
+	for _, f := range fields {
+		b := []byte{}
+
+		if len(table) > 0 {
+			b = append(b, table...)
+			b = append(b, '.')
+		}
+		b = append(b, f.SQLName...)
+		list = append(list, b)
+	}
+	return list
+}
+
+func appendMultiValues(
+	fmter schema.Formatter, b []byte, v reflect.Value, index []int, baseFields, joinFields []*schema.Field, table schema.Safe,
+) []byte {
+	// This is a mix of appendChildValues and query_base.appendColumns
+	if len(joinFields) != len(baseFields) {
+		panic("asdfasdf")
+	}
+
+	// First get the columns
+	joins := getColumns(table, joinFields)
+	// Then values
+	b = append(b, '(')
+	seen := make(map[string]struct{})
+	walk(v, index, func(v reflect.Value) {
+		start := len(b)
+		for i, f := range baseFields {
+			if i > 0 {
+				b = append(b, " AND "...)
+			}
+			if len(baseFields) > 1 {
+				b = append(b, '(')
+			}
+			b = append(b, joins[i]...)
```

Comment:
Looks like we can get rid of `getColumns` and instead write here:

```go
		b = append(b, joinTable...)
		b = append(b, '.')
		b = append(b, joinFields[i].SQLName...)
```

What do you think?